packages/neos-ui-editors/src/Editors/Button/index.js

```diff
@@ -0,0 +1,157 @@
+        multiple: false,
+        disabled: false
+    };
+
+    state = {
+        active: [],
+        isLoading: false,
+        buttons: []
+    };
+
+    initialValueType = 'string';
+
+    constructor(props) {
+        super(props);
+
+        const {commit, options, value} = props;
+        this.initialValueType = Array.isArray(value) ? 'array' : 'string';
+        if (options.multiple && this.initialValueType === 'string') {
+            console.warn(`Misconfiguration in property "${props.identifier}". Multiple is activated but value type seems to be "string" but should be "array".`);
+        }
+
+        this.state = {
+            active: Array.isArray(value) ? value : (value ? [value] : []),
+            buttons: this.hasDataSource() ? [] : this.flattenValues(options.values)
+        };
+        this.commit = commit;
+    }
+
+    hasDataSource() {
+        return this.props.options.dataSourceIdentifier || this.props.options.dataSourceUri;
+    }
+
+    componentDidMount() {
+        if (this.hasDataSource()) {
+            this.loadOptions();
+        }
+    }
+
+    loadOptions() {
+        this.setState({isLoading: true});
+        this.props.dataSourcesDataLoader.resolveValue(getDataLoaderOptionsForProps(this.props), this.props.value)
+            .then(buttons => {
+                this.setState({
+                    isLoading: false,
+                    buttons
+                });
+            });
+    }
+
+    flattenValues(values) {
+        return Object.entries(values).map(([key, val]) => {
+            return {...val, value: key, label: this.props.i18nRegistry.translate(val.label)};
+        });
+    }
+
+    toggleActive(toggleValue) {
+        const {multiple, allowEmpty} = this.props.options;
+        // we'd like to have a copy instead of ref
+        let active = [...this.state.active];
+        if (multiple === false) {
+            if (active.includes(toggleValue)) {
+                active = [];
+            } else {
+                active = [toggleValue];
+            }
+        } else if (active.includes(toggleValue)) {
+            active.splice(active.indexOf(toggleValue), 1);
+        } else {
+            active.push(toggleValue);
+        }
+
+        // if allowEmpty is false but new active length will be 0, drop changes
+        if (!allowEmpty && active.length === 0) {
+            return;
+        }
+        this.setState({active});
+
+        this.commit(this.initialValueType === 'string' ? active[0] || '' : active);
+    }
+
+    isActive(val) {
+        return this.state.active.includes(val);
+    }
+
+    render() {
+        const options = Object.assign({}, this.constructor.defaultOptions, this.props.options);
+
+        return (<div className={style.buttonEditor}>
+            {
+                this.state.isLoading
+                    ? <Icon icon="spinner" size="lg" spin/>
+                    : this.state.buttons.map((button) => (<IconButton
+                            style="lighter"
```

Comment:
```suggestion
                    : this.state.buttons.map((button) => (<IconButton
                            key={button.value}
                            style="lighter"
```

to prevent the following error:

```
Warning: Each child in a list should have a unique "key" prop.

Check the render method of `ButtonEditor`. See https://fb.me/react-warning-keys for more information.
```